torch/csrc/jit/serialization/python_print.cpp

```diff
@@ -1162,23 +1162,67 @@ struct PythonPrintImpl {
         // calculate how many args are specified.
         // see (https://github.com/pytorch/pytorch/pull/56079) for more
         // details.
-        size_t necessary_args =
-            CalculateNecessaryArgs(schema.arguments(), node->inputs());
-        for (const auto i : c10::irange(necessary_args)) {
-          if (i > 0)
+        size_t num_schema_args = schema.arguments().size();
+
+        // we only want to do this extra logic only when necessary.
+        if (num_schema_args > 0) {
+          // calculate how many args are specified.
+          // see (https://github.com/pytorch/pytorch/pull/56079) for more
+          // details.
+          auto specified_args =
+              CalculateNecessaryArgs(schema.arguments(), node->inputs(), true);
+
+          auto necessary_args = specified_args.first + specified_args.second;
+
+          size_t schema_idx = num_schema_args - 1;
+          while (schema_idx >= 0) {
+            auto current_arg = schema.arguments().at(schema_idx);
+            if (!current_arg.is_out()) {
+              break;
+            }
+            schema_idx--;
+          }
+
+          auto num_out = schema.arguments().size() - schema_idx - 1;
```

Comment:
Is this different from `specified_args.second`?

Reply:
hhha good catch. It is the same thing